### DISCUSSION TITLE
Para um iniciante por onde começar Node ou PHP?

### DISCUSSION THREAD

**Discussion by brunoak** [frontendbr/forum#1537] [Discussão]
pessoal estou estudando html e css, e já consigo fazer uma página, agr vou para o próximo passo, vejo que a maioria dos sites usam PHP, mas vejo muitas pessoas falarem do Node também.
Para um iniciante qual seria o caminho mais "facil" em relação a vagas e salario?

**Comment by rafaelangical** [comment]
Recomendo Node, pois já aprende Javasript, dai fica mais fácil aprender os frameworks front-end mais usados hj em dia com React, Vue, Angular, etc, caso queira se tornar Fullstack
.

**Comment by lucasferreiralimax** [comment]
Para iniciante todo conhecimento é válido.

Eu indico uma Stack JavaScript se for seguir a carreira de Front-end Developer

Se já domina o básico da linguagem JavaScript, comece a aprender os frameworks de trabalho que padroniza o desenvolvimento para suas aplicações.

//Front-end
[Vue.js](https://vuejs.org)
[Angular](https://angular.io)
[React.js](https://pt-br.reactjs.org)

Node.js é essencial para um bom Front-end vai te ajudar muito, pode começar conhecendo os pacotes do node no npm e como usa-los

// WorkFlow
[Webpack](https://webpack.js.org)
[Gulp.js](https://gulpjs.com)

// Back-end
[Express.js](https://expressjs.com)
[Nuxt.js](https://nuxtjs.org)

De inicio já é bastante conteúdo para estudar e aprender

Tem esse projeto bem legal mostrando uma trilha de conteúdo para Front-end, Back-end e DevOps.
[Developer Roadmap](https://github.com/kamranahmedse/developer-roadmap)

Bons estudos 👍

**Comment by HmmerHead** [comment]
Dou razão ao o que o pessoal diz, mas veja PHP também. Como disseram a pouco, conhecimento nunca é demais, e inclusive pode ser útil em alguma vaga de emprego por ex. Vejo diversas vagas pedindo os dos inclusive. Aquela que mais te interessar, você vai se aprofundando, porém, não deixando e lado a outra.

**Comment by Vitor-Felipe** [comment]
Concordo com o @HmmerHead e o que o pessoal disse sobre conhecimento, porém no inicio eu acredito que o melhor que você pode fazer é focar em uma linguagem para entender os processos e não a linguagem em si.
Digo isso porque no meu inicio eu estudava varias coisas ao mesmo tempo e acabava tendo um baixo retorno em relação a conhecimento, pois acabava aprendendo os mesmo processos porém em diferentes sintaxes e também me confundia entre elas.

Em relação as linguagens/tecnologias citadas:

PHP:
- Tem muito mercado no Brasil (Não sei como está o mercado fora do Brasil)
- Tem MUITO conteudo na web
- Tem muitos sistemas na web utilizando PHP (tem reportagens que fala que 80% da web tem PHP)
- Curva de aprendizado mais facil que Node.js

Node.js:
- Tem muito mercado no Brasil e fora
- Utiliza JavaScript
- Tem bastante conteudo na web
- Tu consegue utilizar para uma aplicação simples como um to do list, até mesmo um streaming estilo netflix
- Curva de aprendizado maior em comparação com PHP

Minha opnião:
Eu escolheria Node, pois como o @lucasferreiralimax e o @rafaelangical  disseram, você vai estar trabalhando com JS em todas as partes do seu sistema, o que faz com que você aprenda mais rapido e reutilize tecnicas de js no front e back :)

**Reply by Miguel-EpicJS** [reply]
obs: Wordpress é feito em PHP, isso da uma baita alavancagem(você pode vender temas, plugins, etc)

**Reply by miscapu** [reply]
Opa!!..eu trabalho com temas e plugins WordPress; conheço PHP bastante. Consegue me informar algum lugar(forum, site, grupo whatsapp, telegram, etc) onde consiga freelas?
Estive trabalhando para um rapaz que morava no RJ e ele informou que tinha contatos com pessoas no themeforest porém eu não tenho mais comunicação com ele atualmente. Gostaria de entrar na área, agradeço!!

**Comment by brunoak** [comment]
@rafaelangical @lucasferreiralimax  @HmmerHead @Vitor-Felipe Obrigado pelas respostas.
Eu faço curso técnico, e nesse semestre estou vendo PHP, mas aquele bem basicão, e por fora estou vendo JS, e o motivo dessa issue foi bem o que o @Vitor-Felipe  disse, estudar php e js a fundo ao mesmo tempo poderia atrapalhar, seguido o comentario de vcs acho que o momento é focar no JS mesmo

**Comment by MauroSergio61** [comment]
Estou pensando em focar em desenvolvimento Backend também, escolhi Javascript e PHP e estou pensando em estudar as linguagens para depois passar para os frameworks. Quais os tópicos vocês julgam essenciais para estudar ??(já tenho um conhecimento inicial sobre lógica de programação em Java).

**Comment by CLucasrodrigues22** [comment]
Constantimente vem essa duvída na minha cabeça. Terminei meu curso técnico a 5 meses, e lá, vi php OO e PDO, e um pouco de js no front end, e até hoje estou focado em php, mais tenho medo de ficar ultrapassado (se é que essa é a palavra certa) no mercado, já que muitas empresas grandes estão utilizando no back-end o Node.js. Por isso fico com essa questão na cabeça, "será que vale apena continuar focando 100% em php ou devo partir pro Node.js o mais rápido possível?".

**Reply by wilsonneto-swe** [reply]
Slack é feito em PHP e acabou de ser vendido por bilhões, PHP ainda está vivo, mas faça uma pesquisa de vagas e salários em sua cidade/região, e veja, cada região tem suas particularidades, para começar é melhor focar na que tem muita demanda

**Reply by Miguel-EpicJS** [reply]
PHP lançou a versão 8 faz pouco tempo, não estudo PHP mas pelo que sei está longe de morrer...

**Reply by CLucasrodrigues22** [reply]
> PHP lançou a versão 8 faz pouco tempo, não estudo PHP mas pelo que sei está longe de morrer...

Esse papo de PHP morrer já faz muito tempo, eu já ouvia isso antes mesmo de entrar na linguagem, e está ai, versão 8 lançada, continnua sendo a linguagem mais usada para back-end no mundo. Eu acredito que sim, ele pode até perder esse posto, mais creio que vai ser tipo o que o COBOL é hoje, tem muitas vagas para programadores COBOL aqui em Brasília, mais não para criar novos sistemas bancários, mas para da suporte nos sistemas atuais, pois nem todas as empresas tem condição para contratar programadores especializados em tecnólogias atuais e e até mesmo bancar uma infra melhor, esse é o meu ponto de vista.

**Reply by miscapu** [reply]
Eu também escutei muito contra para o PHP; hoje continuo aprendendo; só não estou conseguindo muitos projetos para trabalhar. Não sei o que está acontecendo; realmente em termos de vagas: JS ganha hoje!

**Reply by CLucasrodrigues22** [reply]
Rapaz, nem lembrava mais dessa discussão aqui hahahah, hoje atuo como pleno, 10k trabalhando com symfony. Por esta no início, tinha a visão de mercado muito equivocada, não mudaria nada nas escolhas, a atualmente estudo também Java e spring boot, almejo vaga em algum banco. PHP tem muito mercado, ainda mais com o Laravel, as novas features que são lançadas a cada atualização na linguagem, e na época, eu acreditava no conto de que o PHP morreu, mas a verdade é que muita coisa boa ainda vai vir.

**Comment by francysreymer** [comment]
Eu sou desenvolvedor PHP há mais de 10 anos, e digo que se pudesse voltar
atrás estudaria mais outras linguagens e principalmente javascript (backend
e frontend) quando surgiu.
Digo isso em questão salarial pois javascript além de ter bem mais vagas
vejo que remunera melhor com relação a PHP.






---
Francys Reymer
Engenheiro de Computação | Mestre em Engenharia Elétrica
Fone/Whatsapp: (67) 98429-5618

**Comment by denisjuvino** [comment]
Minha opinião é que é sempre melhor, quando se está iniciando, escolher apenas uma linguagem e se aprofundar nela, em seus melhores frameworks, e tambem banco de dados SQL, Mysqli e etc. É melhor focar em ser especialista em uma linguagem do que ser intermediário em duas ou três. O PHP por exemplo tem o Laravel, se você se especializar em PHP, Laravel, banco de dados e Apis, poderá ser um bom desenvolvedor Backend. Seja bom, muito bom mesmo em uma coisa, pra depois partir pra outra. Dei exemplo em PHP porque me identifico com a linguagem, não ligo por salários mais altos (visto que PHP também paga muito bem) ou maior quantidade de vagas, (PHP tem muitas vagas no mercado, talvez até mais que Node), e sim porque amo o que faço, amo ser desenvolvedor, e amo o PHP.

**Reply by miscapu** [reply]
Se vc compartilhar algum canal para encontrar freelas usando a linguagem PHP; seria excelente. Hoje procuro muito porém tem muitas empresas focando em JS.
